### PR TITLE
New Simulator 

### PR DESCRIPTION
**Pull request by babusid**
This PR contains the new stm32f413 simulator built on Renode, customized for Controls usage. It contains three major components:
- A BPS Simulator
- Controls Simulator
- Motor Simulator

The Motor Simulator and BPS simulator are currently pre-compiled test files. The Motor will remain this way, however, the BPS simulator in the future could become a faithful representation of the actual BPS system. This will require additional peripheral development. 

The Controls simulator is currently running the same .elf file as the actual system. We are aiming to simulate all input/output behavior of the system in an extensible manner, so as to allow for quick recreations of future control interfaces.

### CODE REVIEW

File: Simulator/startup.resc [hunk 1]
```diff
@@ -0,0 +1,93 @@
+$bin?=@Objects/controls-leader.elf
+$bps_bin?=@Objects/car-sim.elf
+$motor_bin?=@Objects/motor-sim.elf
+$platform=@Simulator/stm32f413.repl
+
+macro reset
+"""
+    sysbus LoadELF $bin
```
Comment (NathanielDelgado) [inline Simulator/startup.resc:8]: does reset actually reset the emulation by just loading in the elf again? I would think that `Clear` does the same

File: CarSim/car-sim.c
Comment (NathanielDelgado) [inline CarSim/car-sim.c:1]: This should be deleted or at least not called car-sim.c

File: MotorSim/motor-sim.c
Comment (NathanielDelgado) [inline MotorSim/motor-sim.c:1]: This should not be called motor-sim or should be deleted.

File: Simulator/startup.resc [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+"""
+    emulation CreateServerSocketTerminal 49154 "motor-uart-con" false
+    connector Connect sysbus.usart2 motor-uart-con
+"""
+
+macro show_uart_bps
+"""
+    emulation CreateServerSocketTerminal 49155 "bps-uart-con" false
+    connector Connect sysbus.usart2 bps-uart-con
+"""
```
Comment (NathanielDelgado) [inline Simulator/startup.resc:33]: Take out `-con` on the names (more typing for no reason)

File: Simulator/startup.resc [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+print 'Offset: %s'%hex(offset)
+print "\n"
+"""
+
+set periphReadHook """
+print "START READ HOOK"
+print 'Periph: %s'%self
+print 'Value: %s'%hex(value)
+print 'Offset: %s'%hex(offset)
+print "\np"
```
Comment (NathanielDelgado) [inline Simulator/startup.resc:48]: \np -> \n

**Review by IshDeshpa — APPROVED** [review]
LGTM. Some stuff like ReadCarCAN test file will be overwritten after ReadCarCAN+newsim is merged.

**Review by Cam0Cow — APPROVED** [review]

**Review by NathanielDelgado — APPROVED** [review]

**Review by babusid — COMMENTED** [review]
let's do dis

**Pull request merged**